app/src/main/java/com/android/streetworkapp/model/image/ImageRepositoryFirestore.kt

```diff
@@ -270,21 +275,12 @@ class ImageRepositoryFirestore(
       for (document in querySnapshot.documents) {
         val collection = document.toObject(ParkImageCollection::class.java)
         collection?.let {
-          val (imagesUploadedByUser, imagesNotUploadedByUser) =
-              collection.images.partition { it.userId == userId }
-          if (imagesUploadedByUser.isNotEmpty())
-              document.reference
-                  .update("images", FieldValue.arrayRemove(imagesUploadedByUser))
-                  .await()
-
-          imagesUploadedByUser.forEach {
-            val imageKey = this.storageClient.extractKeyFromUrl(it.imageUrl) ?: return false
-            if (!this.storageClient.deleteObjectFromKey(imageKey)) return false
-          }
-
-          // remove any potential votes the user has done
-          for (image in imagesNotUploadedByUser) {
-            if (image.rating.positiveVotesUids.contains(userId) ||
+          for (image in collection.images) {
+            if (image.userId == userId) {
+              document.reference.update("images", FieldValue.arrayRemove(image)).await()
+              val imageKey = storageClient.extractKeyFromUrl(image.imageUrl) ?: return false
+              if (!this.storageClient.deleteObjectFromKey(imageKey)) return false
+            } else if (image.rating.positiveVotesUids.contains(userId) ||
```

Comment:
SonarCloud raised an issue here suggesting to "_Merge this ‘if’ statement with the nested one_" to simplify the code structure.

Reply:
I'll keep it as is, I think it makes the flow clearer as is, would have a very ugly condition loop otherwise

Reply:
I agree with you, it would be a too big condition. I'll leave you to accept the issue in SonarCloud [here](https://sonarcloud.io/project/issues?sinceLeakPeriod=true&issueStatuses=OPEN%2CCONFIRMED&pullRequest=237&id=SwEnt-Group8_Street-work-app&open=AZPVTIkBAdeY80wzHnWF).